toplevel/byte/topmain.ml

```diff
@@ -178,7 +159,7 @@ let input_argument name =
       in
       Compenv.readenv ppf Before_link;
       Compmisc.read_clflags_from_env ();
-      if prepare ppf && Toploop.run_script ppf name newargs
+      if Toploop.prepare ppf && Toploop.run_script ppf name newargs
```

Comment:
The `run_script` function is reinitializing the environment, I believe it would be better to merge the two initialization (by removing the initialization code in `run_script`.